### DISCUSSION TITLE
Starting New

### DISCUSSION THREAD

**Discussion by mdavidovichHst** [intersystems/git-source-control#186] [Q&A]
Got everything setup and working.  Was able add/track a file no problem.  Thought the next step was to Add everything from my app so that I could have it in the local file system and then push to the remote to.  Assume that there's no existing remote repository and this is the first time we are pushing the app to a remote to enable better dev practices.  

I right clicked the top level 'isfs' for the namespace and did the Add.  Got this error:

```
Executing Studio Action "Add" on localhost:52773[clariti] failed with the following error:

Non-JSON response to /api/atelier/v4/CLARITI/action/query request. Is the web server suppressing detailed errors?
```

A few things I noticed:
- The code was exported to my local file system in a very different way than what you see in VS Code or Studio.  Classes were all under a single folder ('cls/<package>/<class>')
- The same was for INC files.  
- I was surprised to the the CSP files were pulled too ('csp/csp/<webApp>/<cspFIle>).  Does this mean I can edit the CSP files using isfs and when changes are added they'll export to this path?  
- I assume if it hadn't errored, I would have got my MAC files too.

Other Questions:
- How can I get the entire project committed without failing so I can push it to the empty remote?
- When I pull from the menu in VS Code, I assume it knows how to put the packages in the right places versus how they are stored locally?
- This is very different from the client side method of working on code where you download it all locally and as you change and save, objecscript.conn pushes it back to the server.  The client side method seems more native to how it's all built, but this server-side first method makes more sense but is not technically supported by InterSystems as an open exhange project (I know Tim wrote it though). 
- What about packages like 'Ens' that are part of InterSystems IRIS for Health and you don't need or want to track them?  How do you add them to the .gitignore file?  
- I guess in general, since VSCode doesn't support projects like Studio did, how do you ignore all the system stuff like Ens or EnsHL7?

Lots to discuss but I have learned that if I'm asking then there's probably someone else asking too.  I hope this is helpful to other's trying this out.

**Comment by isc-tleavitt** [comment]
I think the answer to a lot of your questions comes down to configuring mappings for server-side name to location on disk. I've never tried "Add" on the top-level isfs folder - but suspect that if you added more granularly, that would work just fine.

Among server-side source control extensions for InterSystems' platforms, there has been a convention in the past of separating files by type (/cls for classes, /rtn for routines, /inc for include files). You can customize this using mappings in the extension configuration. Through this approach, certain packages may be mapped - you wouldn't map (or track) Ens/EnsHL7/etc, just your own code.

**Reply by isc-tleavitt** [reply]
Also - yes, you can manage CSP resources over isfs with proper behavior on import/export, although I believe @isc-svelury has done some relevant work on that which will be included in the upcoming 2.0.0 release (later this month).
